internal/operands/vm-console-proxy/reconcile_test.go

```diff
@@ -35,49 +38,187 @@ const (
+	})
+
+	It("should not update service cluster IP", func() {
+		_, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		key := client.ObjectKeyFromObject(&bundle.Service)
+		service := &core.Service{}
+		Expect(request.Client.Get(request.Context, key, service)).ToNot(HaveOccurred())
+
+		// This address is from a range of IP addresses reserved for documentation.
+		const testClusterIp = "198.51.100.42"
+
+		service.Spec.ClusterIP = testClusterIp
+		Expect(request.Client.Update(request.Context, service)).ToNot(HaveOccurred())
+
+		_, err = operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		updatedService := &core.Service{}
+		Expect(request.Client.Get(request.Context, key, updatedService)).ToNot(HaveOccurred())
+		Expect(updatedService.Spec.ClusterIP).To(Equal(testClusterIp))
+	})
+
+	It("should report status", func() {
+		reconcileResults, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Set status for deployment
+		key := client.ObjectKeyFromObject(&bundle.Deployment)
+		updateDeployment(key, &request, func(deployment *apps.Deployment) {
+			deployment.Status.Replicas = replicas
+			deployment.Status.ReadyReplicas = 0
+			deployment.Status.AvailableReplicas = 0
+			deployment.Status.UpdatedReplicas = 0
+			deployment.Status.UnavailableReplicas = replicas
+		})
+
+		reconcileResults, err = operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		// Only deployment should be progressing
+		for _, reconcileResult := range reconcileResults {
+			if _, ok := reconcileResult.Resource.(*apps.Deployment); ok {
+				Expect(reconcileResult.Status.NotAvailable).ToNot(BeNil())
+				Expect(reconcileResult.Status.Progressing).ToNot(BeNil())
+				Expect(reconcileResult.Status.Degraded).ToNot(BeNil())
+			} else {
+				Expect(reconcileResult.Status.NotAvailable).To(BeNil())
+				Expect(reconcileResult.Status.Progressing).To(BeNil())
+				Expect(reconcileResult.Status.Degraded).To(BeNil())
+			}
+		}
+
+		updateDeployment(key, &request, func(deployment *apps.Deployment) {
+			deployment.Status.Replicas = replicas
+			deployment.Status.ReadyReplicas = replicas
+			deployment.Status.AvailableReplicas = replicas
+			deployment.Status.UpdatedReplicas = replicas
+			deployment.Status.UnavailableReplicas = 0
+		})
+
+		reconcileResults, err = operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		// All resources should be available
+		for _, reconcileResult := range reconcileResults {
+			Expect(reconcileResult.Status.NotAvailable).To(BeNil())
+			Expect(reconcileResult.Status.Progressing).To(BeNil())
+			Expect(reconcileResult.Status.Degraded).To(BeNil())
+		}
+	})
+
+	It("should deploy resources in namespace provided by annotation", func() {
+		_, err := operand.Reconcile(&request)
+		Expect(err).ToNot(HaveOccurred())
+
+		namespace := request.Instance.GetAnnotations()[VmConsoleProxyNamespaceAnnotation]
```

Comment:
Instead of getting the namespace, can you change the value of the annotation to something different?
And then after calling `operand.Reconcile()`, test that the resources are deployed in that namespace.

Reply:
Since I'm doing: `ExpectNamespacedResourceExists(&bundle.Deployment, request, namespace)` it doesn't work, I think this is because `&bundle.Deployment` contains value of `namespace` from constant variable.

Reply:
Then we should do: `ExpectNamespacedResourceExists(operand.serviceAccount, request, namespace)` to use changed resource in `operand.serviceAccount`?

Reply:
I'm trying to do it, but for some reason it doesn't work, I'm spending some time to debug it.

Reply:
Done, thanks @akrejcir.